atlasdb-dbkvs-hikari/src/main/java/com/palantir/nexus/db/pool/config/ConnectionProtocol.java

```diff
@@ -28,9 +30,10 @@ public String getUrlString() {
         return urlStr;
     }
 
+    @JsonCreator
     public static ConnectionProtocol fromUrlString(String val) {
         for (ConnectionProtocol cp : ConnectionProtocol.values()) {
-            if (cp.getUrlString().toLowerCase().equals(val.toLowerCase())) {
+            if (cp.getUrlString().equalsIgnoreCase(val)) {
```

Comment:
While I agree that `equalsIgnoreCase` is better than what we were doing before, it _should_ be semantically the same - and indeed, reverting this change resulted in the tests passing.

I think the fix here is actually `@JsonCreator` if I am not mistaken - before, it was trying to map directly to one of the enums, whereas now it actually goes through this method - is that right?

Reply:
yes, this is just a slight optimization, as well as making the code more readable

Reply:
it also avoids an extra string allocation for the case that val is not already lowercase